server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -64,13 +58,25 @@ public static <K, V> Map<K, V> copyMapWithAddedEntry(final Map<K, V> map, final
      * @param <V>   the type of the values in the map
      * @return an immutable map that contains the items from the specified map and a mapping from the specified key to the specified value
      */
+    @SuppressWarnings("unchecked")
     public static <K, V> Map<K, V> copyMapWithAddedOrReplacedEntry(final Map<K, V> map, final K key, final V value) {
```

Comment:
Changes here aren't 100% necessary for this to work, but they cause the whole thing to compile far better.
Without this change there's not really any reason to improve this code though so I put it in here.